arviz/stats/stats.py

```diff
@@ -439,15 +440,11 @@ def loo(data, pointwise=False, reff=None, scale="deviance"):
 
     """
     inference_data = convert_to_inference_data(data)
-    for group in ("posterior", "sample_stats"):
-        if not hasattr(inference_data, group):
-            raise TypeError(
-                "Must be able to extract a {group} group from data!".format(group=group)
-            )
-    if "log_likelihood" not in inference_data.sample_stats:
-        raise TypeError("Data must include log_likelihood in sample_stats")
+    if not hasattr(inference_data, "posterior"):
+        raise TypeError("Must be able to extract a posterior group from data!")
     posterior = inference_data.posterior
-    log_likelihood = inference_data.sample_stats.log_likelihood
+    log_likelihood = _get_log_likelihood(inference_data)
```

Comment:
I believe that this must be changed to 
```
log_likelihood = _get_log_likelihood(inference_data, "lp")
```
Otherwise I get an error doing `az.compare` or `az.waic` on a dataset with multiple observed variables -- get log likelihood complains that its `var_name` argument cannot be none if there are multiple variables in the `log_likelihood` group of the `InferenceData`.  I believe in this case one wants the composite log-likelihood from the special `lp` variable.

Reply:
We do have to look into this, I am not sure what would be correct in these cases with multiple variables though. Another possibility would be to calculate loo/waic for each variable or simply to stack together all variables in `log_likelihoods` except `lp`. Intuitively, using `lp` feels weird as you have the same value for all observations, it only depends on the draw and chain, so I would not know how to interpret the pointwise loo/waic values.

I remember discussing this at some point with @ahartikainen and @aloctavodia, maybe you have had time to look into this? I still have not been able to work out the maths nor experiment at all.